index.js

```diff
@@ -327,71 +318,131 @@ if(!helpers.isTestEnvironment()){
-    const {deviceid, locationid, devicetype, state, rpm, distance, mov_f, mov_s} = req.body;
-
-    redis.addXeThruSensorData(req, res);
-    handleSensorRequest(locationid);
-});
-
-app.post('/api/doorTest', async(req, res) => {
-    const {locationid} = req.body;
-    await redis.addDoorTestSensorData(req, res);
-    await handleSensorRequest(locationid)
+app.post('/api/xethru', Validator.body(['locationid', 'state', 'rpm', 'distance', 'mov_f', 'mov_s']).exists(), async (req, res) => {
+    try {
+        const validationErrors = Validator.validationResult(req)
+
+        if(validationErrors.isEmpty()){
+            // eslint-disable-next-line no-unused-vars -- might be useful in the future to know what all we have access to in the body
+            const { deviceid, locationid, devicetype, state, rpm, distance, mov_f, mov_s } = req.body;
+
+            await redis.addXeThruSensorData(locationid, state, rpm, distance, mov_f, mov_s);
+            await handleSensorRequest(locationid);
+            res.status(200).send()
+        } else {
+            helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
+            res.status(400).send()
+        }
+    } catch (err) {
+        helpers.log(err)
+        res.status(500).send()
+    }
 })
 
-app.post('/api/door', async(request, response) => {
-    const coreId = request.body.coreid
-    const locationid = await db.getLocationIDFromParticleCoreID(coreId)
+app.post('/api/doorTest', Validator.body(['locationid', 'signal']).exists(), async(req, res) => {
+    try {
+        const validationErrors = Validator.validationResult(req)
 
-    if(!locationid){
-        helpers.log(`Error - no location matches the coreID ${coreId}`)
-        response.status(400).json('No location for CoreID')
-    }
-    else{
-        const message = JSON.parse(request.body.data)
-        const signal = message.data
-        var doorSignal
-        if (signal==IM21_DOOR_STATUS.OPEN){
-            doorSignal = SESSIONSTATE_DOOR.OPEN
-        }
-        else if (signal==IM21_DOOR_STATUS.CLOSED){
-            doorSignal = SESSIONSTATE_DOOR.CLOSED
-        }
-        else if (signal==IM21_DOOR_STATUS.LOW_BATT){
-            doorSignal = "LowBatt"
-        }
-        else if (signal==IM21_DOOR_STATUS.HEARTBEAT_OPEN || signal==IM21_DOOR_STATUS.HEARTBEAT_CLOSED){
-            doorSignal = "HeartBeat"
+        if(validationErrors.isEmpty()){
+            const { locationid, signal } = req.body
+
+            await redis.addDoorTestSensorData(locationid, signal)
+            await handleSensorRequest(locationid)
+            res.status(200).send()
+        } else {
+            helpers.log(`Bad request, parameters missing ${JSON.stringify(validationErrors)}`)
+            res.status(400).send()
         }
+    } catch (err) {
+        helpers.log(err)
+        res.status(500).send()
+    }
+})
+
+app.post('/api/door', Validator.body(['coreid']).exists(), async(request, response) => {
```

Comment:
as noted in our call could be a good idea to check for the fields we use that it gets from request.body.data

Reply:
Updated :)

Reply:
looks good! this is good to go then

Reply:
I think I'd prefer a squash commit with a nice message? @trisapeace you agree?

Reply:
Haha yeah that was definitely my plan cause it's a bit of a mess on that front